### PR TITLE
DOC-757 | Edge tutorial 

### PR DESCRIPTION
**Pull request by sudkul87**
## Describe the Change

This PR proposes enhancements to the original tutorial written by Justin Barksdale in [PR#1286](https://github.com/spectrocloud/librarium/pull/1286). 

## Review Changes

💻 [Preview URL](https://deploy-preview-1291--docs-spectrocloud.netlify.app/clusters/edge/deploy-cluster)

🎫 Jira Ticket [DOC-757](https://spectrocloud.atlassian.net/browse/DOC-757). This Jira ticket duplicates [DOC-758](https://spectrocloud.atlassian.net/browse/DOC-758)

[Tutorial Plan document](https://docs.google.com/document/d/1gSuwwD7QhQnkMs8Yu6dyuoEjYJT_LpP0GIAbDq3laIs/edit?usp=sharing)

### CODE REVIEW

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1291#" title="Author self-approved">sudkul87</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1291#" title="Author self-approved">sudkul87</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/DOC-758-edge-tutorial-for-v-mware/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1291#" title="Author self-approved">sudkul87</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **zulfilee**
You can assign the PR to them by writing `/assign @zulfilee` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/DOC-758-edge-tutorial-for-v-mware/OWNERS)**
- **[prow/OWNERS](https://github.com/spectrocloud/librarium/blob/DOC-758-edge-tutorial-for-v-mware/prow/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1291#" title="Author self-approved">sudkul87</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/release-3-4/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: content/docs/04-clusters/03-edge/11-deploy-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/11-deploy-cluster.md:35]: This diagram - cloning a template does not feel like the right word. Packer creates the template, and then GOVC is used to deploy an instance using the template as a blueprint.  So to me, step 5 is deploying VMs.
CleanShot 2023-06-16 at 14 00 09

File: content/docs/04-clusters/03-edge/11-deploy-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/11-deploy-cluster.md:405]:
```suggestion
GOVC requires the same VMWare vCenter details as the environment variables you defined earlier in the **.packerenv** file. Use the following command to source the **.packerenv** file and echo one of the variables to ensure the variables are accessible on your host machine. 
```

File: content/docs/04-clusters/03-edge/11-deploy-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/11-deploy-cluster.md:413]:
```suggestion
Use the following command to create an environment file titled **.goenv**. The  **.goenv** file contains the required VMWare vCenter credentials and information required to deploy VMs in your VMware environment. 
```

Simplified the langauage.

File: content/docs/04-clusters/03-edge/11-deploy-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/11-deploy-cluster.md:431]:
```suggestion
Issue the following command to deploy three VMs in the VMWare vCenter. 
```
Let's keep it simple. The user wants to create the instances.

File: content/docs/04-clusters/03-edge/11-deploy-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/11-deploy-cluster.md:435]: Same feedback as earlier, break up the command to two or three lines. Same feedback about `--cpu-shares`

File: content/docs/04-clusters/03-edge/11-deploy-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/11-deploy-cluster.md:465]: Same as earlier, if you're gonna mention this then do it before the command that uses this file.

File: content/docs/04-clusters/03-edge/11-deploy-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/11-deploy-cluster.md:488]: Same feedback as earlier. What is out of scope?

File: content/docs/04-clusters/03-edge/11-deploy-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/11-deploy-cluster.md:493]:
```suggestion
After building the Edge artifacts and provisioning VMs as Edge hosts, the next step is to verify the Edge host registration status, create a cluster profile, and deploy a cluster that is made up of the Edge hosts. 
```

File: content/docs/04-clusters/03-edge/11-deploy-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/11-deploy-cluster.md:497]:
```suggestion
Open a web browser and log in to [Palette](https://console.spectrocloud.com). Navigate to the left **Main Menu** and select **Clusters**. Click on the **Edge Hosts** tab and verify the three VMs created are registered with Palette.
```
1. Let's be optimistic and confident in our product. 
2. This is the first time the user is doing something in Palette, so we need to tell them to log in.
3. Don't take shortcuts. Help the user find the right place in the UI by guiding with sentences.

File: content/docs/04-clusters/03-edge/11-deploy-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/11-deploy-cluster.md:502]:
```suggestion
If the three Edge hosts are not displayed in the list then the automatic registration failed. You can register the Edge hosts manually by using each Edge host's ID. Click on the **Add Edge Hosts** button, and paste the Edge host ID returned after provisioning the VMs in one of the previous steps. The Edge host IDs can be found in the GOVC output that was generated when you deployed the VMs.
```

Might wa

File: content/docs/04-clusters/03-edge/11-deploy-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/11-deploy-cluster.md:506]:
```suggestion
Validate you are in the **Default** project scope before creating a cluster profile.  
```

File: content/docs/04-clusters/03-edge/11-deploy-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/11-deploy-cluster.md:513]:
```suggestion
```
Let's remove this and combine with the next line. Also, the order is incorrect, mention the left main menu, then the profiles section.

File: content/docs/04-clusters/03-edge/11-deploy-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/11-deploy-cluster.md:516]:
```suggestion
Next, create a cluster profile comprising the core infrastructure layers and a manifest of a sample application, [Hello Universe](https://github.com/spectrocloud/hello-universe#hello-universe). 
Navigate to the left **Main Menu** and select **Profile**. Click on the **Add Cluster Profile** button, and fill out the required input fields. The cluster profile wizard contains the following sections. 
```

I changed this paragraph by providing the context of what is about, followed by the steps.

File: content/docs/04-clusters/03-edge/11-deploy-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/11-deploy-cluster.md:587]:
```suggestion
Select the K3s version 1.25.x. 1.25.X is used because you pushed a provider image compatible with K3s v1.25.2 to the *ttl.sh* image registry earlier in this tutorial. BYOOS pack's `system.uri` attribute will reference the Kubernetes version you select using the `{{ .spectro.system.kubernetes.version }}` [macro](/clusters/cluster-management/macros).
```

File: content/docs/04-clusters/03-edge/11-deploy-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/11-deploy-cluster.md:590]:
```suggestion
Click on the **Next layer** button, and add the following network layer. This example uses the Calico Container Network Interface (CNI). However, you can choose a different CNI pack that fits your needs, such as Flannel, Cilium, or Custom CNI. 
```

File: content/docs/04-clusters/03-edge/11-deploy-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/11-deploy-cluster.md:600]:
```suggestion
Next, click on the **Add Manifest** button on the top to add the Hello Universe application manifest.  
```

File: content/docs/04-clusters/03-edge/11-deploy-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/11-deploy-cluster.md:613]:
```suggestion
When you provide the value `hello-universe` in the **Manifest** field, a blank text editor will open o the righthand side of the screen. Copy the following manifest and paste it into the text editor. 
```

File: content/docs/04-clusters/03-edge/11-deploy-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/11-deploy-cluster.md:665]: Let's skip this, we don't need to explain the app. That's not one of the objectives for this tutorial.

File: content/docs/04-clusters/03-edge/11-deploy-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/11-deploy-cluster.md:646]:
```suggestion
        image: ghcr.io/spectrocloud/hello-universe:1.0.12 
```
All images are static so it's kind redundant to say so 😅

File: content/docs/04-clusters/03-edge/11-deploy-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/11-deploy-cluster.md:670]:
```suggestion
```
If a section only has a single line let's just do without it.

File: content/docs/04-clusters/03-edge/11-deploy-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/11-deploy-cluster.md:675]:
```suggestion
Click on the newly created cluster profile to view its details page.  Click the **Deploy** button to deploy a new Edge cluster. 
```

The refer statement will cause someone to leave the guide.  Best to remove it. If you want to mention it, use an infobox. But ask yourself first, is it needed? And does it make sense to introduce that here?

File: content/docs/04-clusters/03-edge/11-deploy-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/11-deploy-cluster.md:723]: Let's add a screenshot here too. This is perhaps one of the most important part of the flow so it warrants an image.

File: content/docs/04-clusters/03-edge/11-deploy-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/11-deploy-cluster.md:768]: Why is this screen shot so narrow compared to everything else?

File: content/docs/04-clusters/03-edge/11-deploy-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/11-deploy-cluster.md:831]: Show the the commands to clean the images used in this tutorial. You did it for the docker push command, so let's also do it here.

File: content/docs/04-clusters/03-edge/11-deploy-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/11-deploy-cluster.md:839]: A conclusion should provide a high-level overview  of the key topics a user learned ✅  
The conclusion should also highlight the value Palette provides. Why go through all this? Look to your intro. Remember the scenario you painted of a user deploying Edge hosts to 1000 stores? How can you tie that story into the conclusion to remind the user of Palette's benefits?

I'm going to challenge you to do this in two paragraphs. Be concise and to the point - that's the true writing challenge.

**Review by karl-cardenas-coding — CHANGES_REQUESTED** [review]
Great start, but we have some polishing and clinging up to do.  I also don't see a link to the tutorial plan. My guess is we didn't create one for this tutorial (tsk tsk) but I could be wrong.  COmb through the feedback carefully because of it requires you to think about how the information is presented.

File: content/docs/04-clusters/03-edge/11-deploy-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/11-deploy-cluster.md:750]: Spend some time here explaining that you are using the Edge hosts to make up the cluster nodes. One acts as the control plane node, and the other two as the worker nodes. This is a critical point of the learning journey.

File: content/docs/04-clusters/03-edge/11-deploy-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/11-deploy-cluster.md:790]: Let's add a youTube video here. Justin might be close with his video, otherwise, find an Edge video from our official YouTube channel.

File: content/docs/04-clusters/03-edge/11-deploy-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/11-deploy-cluster.md:1094]: We also need to guide the user in cleaning up the VM template and the uploaded ISO from their folder. This adds up, so make sure you have deleted your templates and ISOs as well.

File: content/docs/04-clusters/03-edge/11-deploy-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/03-edge/11-deploy-cluster.md:1126]:
```suggestion
Palette's Edge solution allows you to prepare your Edge hosts with the desired OS, dependencies, and user data configurations. It supports multiple Kubernetes versions while building the Edge artifacts and creating cluster profiles, enabling you to choose the desired Kubernetes version for your cluster deployment. Before you plan a production-level deployment at scale, you can prepare a small set of Edge devices for development testing and validate the devices' state and installed applications. Once the validation is satisfactory and meets your requirements, you can roll out Edge artifacts and cluster profiles for deployment in production. This approach maintains consistency while deploying Kubernetes clusters at scale across all physical sites, be it 1000 or more sites. In addition to all these benefits, you can conveniently manage the entire lifecycle of Edge clusters with Palette.

To learn more about Edge,  check out the resources below.
```

We don't use please 😉

**Review by karl-cardenas-coding — APPROVED** [review]
Much smoother reading experience this time @sudkul87 . Below are some suggestions and edits. Next up, tag Rita for the final copy edits.

**Review by ritawatson — APPROVED** [review]
Requesting some wording changes. I flagged command options to spell out because that's what we do in the main docs. Good instructive but friendly voice.

**Review by ritawatson — APPROVED** [review]
Requesting some wording changes. I flagged command options to spell out because that's what we do in the main docs. Good instructive but friendly voice.

**Pull request merged**
